src/main/java/data/TaskManager.java

```diff
@@ -33,12 +32,19 @@ public List<String> getTasksForDate(LocalDate date) {
         return tasks.getOrDefault(date, new ArrayList<>());
     }
 
-    public static void addManager(Scanner scanner, WeekView weekView, TaskManager taskManager)
+    public static void addManager(Scanner scanner, WeekView weekView, TaskManager taskManager, boolean inMonthView)
             throws TaskManagerException {
         System.out.println("Enter the date for the task (dd/MM/yyyy):");
         LocalDate date = getStringFromUser(scanner);
 
-        checkIfDateInCurrentWeek(date, weekView);
+        //checkIfDateInCurrentWeek(date, weekView);
+        if (inMonthView) {
+            // Check if the date is within the current month
+            checkIfDateInCurrentMonth(date);
+        } else {
+            // Check if the date is within the current week
+            checkIfDateInCurrentWeek(date, weekView);
+        }
```

Comment:
The comments may be unnecessary, since your code already very effectively describes what is happening.